examples/VLCB4IN4OUT/VLCB4IN4OUT.ino

```diff
@@ -0,0 +1,519 @@
+#else
+#define DEBUG_PRINT(S)
+#endif
+
+// 3rd party libraries
+#include <Streaming.h>
+#include <Bounce2.h>
+
+// VLCB library header files
+#include <Controller.h>                   // Controller class
+#include <CAN2515.h>               // CAN controller
+#include <Switch.h>             // pushbutton switch
+#include <LED.h>                // VLCB LEDs
+#include <Configuration.h>             // module configuration
+#include <Parameters.h>             // VLCB parameters
+#include <vlcbdefs.hpp>               // VLCB constants
+#include <LEDUserInterface.h>
+#include "MinimumNodeService.h"
+#include "NodeVariableService.h"
+#include "EventConsumerService.h"
+#include "EventProducerService.h"
+#include "EventTeachingService.h"
+#include "CanService.h"
+
+#include "LEDControl.h"
+
+////////////DEFINE MODULE/////////////////////////////////////////////////
+
+// module name
+unsigned char mname[7] = { '4', 'I', 'N', '4', 'O', 'U', 'T' };
+
+// constants
+const byte VER_MAJ = 1;     // code major version
+const char VER_MIN = 'a';   // code minor version
+const byte VER_BETA = 0;    // code beta sub-version
+const byte MODULE_ID = 98;  // VLCB module type
+
+const byte LED_GRN = 4;             // VLCB green Uninitialised LED pin
+const byte LED_YLW = 7;             // VLCB yellow Normal LED pin
+const byte SWITCH0 = 8;             // VLCB push button switch pin
+
+// Controller objects
+VLCB::LEDUserInterface userInterface(LED_GRN, LED_YLW, SWITCH0);
+VLCB::Configuration modconfig;               // configuration object
+VLCB::CAN2515 can2515;                  // CAN transport object
+VLCB::MinimumNodeService mnService;
+VLCB::CanService canService;
+VLCB::NodeVariableService nvService;
+VLCB::EventConsumerService ecService;
+VLCB::EventTeachingService etService;
+VLCB::EventProducerService epService;
+VLCB::Controller controller(&userInterface, &modconfig, &can2515, 
+                            { &mnService, &canService, &nvService, &etService, &epService, &ecService }); // Controller object
+
+// Module objects
+const byte LED[] = {3, 5, 6, 9};     // LED pin connections through typ. 1K8 resistor
+const byte SWITCH[] = {A0, A1, A2 , A3}; // Module Switch takes input to 0V.
+
+const byte NUM_LEDS = sizeof(LED) / sizeof(LED[0]);
+const byte NUM_SWITCHES = sizeof(SWITCH) / sizeof(SWITCH[0]);
+
+// module objects
+Bounce moduleSwitch[NUM_SWITCHES];  //  switch as input
+LEDControl moduleLED[NUM_LEDS];     //  LED as output
+byte switchState[NUM_SWITCHES];
+
+// forward function declarations
+void eventhandler(byte, VLCB::CANFrame *);
+void processSerialInput(void);
+void printConfig(void);
+void processModuleSwitchChange(void);
+
+//
+///  setup VLCB - runs once at power on called from setup()
+//
+void setupVLCB()
+{
+  // set config layout parameters
+  modconfig.EE_NVS_START = 10;
+  modconfig.EE_NUM_NVS = NUM_SWITCHES;
+  modconfig.EE_EVENTS_START = 50;
+  modconfig.EE_MAX_EVENTS = 64;
+  modconfig.EE_PRODUCED_EVENTS = 0;
```

Comment:
You have 4 switches that may produce events. Shouldn't EE_PRODUCED_EVENTS be 4 here?